### PR TITLE
completed react-validated-input-component

### PR DESCRIPTION
**Pull request by bbrown949**
https://github.com/bbrown949/c0523-code-solutions/assets/117103446/11762b54-cedf-4f74-847d-1fdf519ed175

### CODE REVIEW

File: react-validated-input-component/my-app/src/ValidatedInput.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  if (password.length >= 0 && password.length < 8) {
+    text = 'Your password is too short.';
+    checkOrX = '❌';
+  }
+  if (password.length === 0) {
+    text = 'A password is required.';
+    checkOrX = '❌';
+  } else if (password.length > 7) {
+    checkOrX = '✅';
+  }
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.js:22]: You could probably combine this into one big `if...else-if...else`

File: react-validated-input-component/my-app/src/ValidatedInput.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <>
+      <label>Password</label>
+      <div>
+        <input
+          type="password"
+          name="password"
+          onChange={(event) => setPassword(event.target.value)}
+          value={password}
+          onSubmit={handleSubmit}></input>
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.js:33]: `'submit'` events only trigger on `<form>`s, not on form controls.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: react-validated-input-component/my-app/src/ValidatedInput.js (hunk 2 above)
Comment (bbrown949) [reply react-validated-input-component/my-app/src/ValidatedInput.js]: fixed

File: react-validated-input-component/my-app/src/ValidatedInput.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  } else if (password.length > 7) {
+    checkOrX = '✅';
+  }
+
+  return (
+    <>
+      <div>
+        <form>
+          <div>
+            <label>Password</label>
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.js:29]: Make sure your `<label>`s are attached to their form controls, either by wrapping them or using the `for` attribute.

File: react-validated-input-component/my-app/src/ValidatedInput.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          </div>
+          <input
+            type="password"
+            name="password"
+            onChange={(event) => setPassword(event.target.value)}
+            value={password}
+            onSubmit={handleSubmit}></input>
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.js:36]: `"submit"` events only trigger on `<form>`s, not `<input>`s. Please move the `onSubmit` handler to the correct element.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: react-validated-input-component/my-app/src/ValidatedInput.js (hunk 4 above)
Comment (bbrown949) [reply react-validated-input-component/my-app/src/ValidatedInput.js]: fixed

File: react-validated-input-component/my-app/src/ValidatedInput.js (hunk 3 above)
Comment (bbrown949) [reply react-validated-input-component/my-app/src/ValidatedInput.js]: fixed

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**